sender-kinesis/src/test/java/zipkin/reporter/kinesis/KinesisSenderTest.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2016-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.reporter.kinesis;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.AnonymousAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.cbor.CBORFactory;
+import java.io.IOException;
+import java.util.List;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+import okio.Buffer;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import zipkin.Codec;
+import zipkin.Span;
+import zipkin.TestObjects;
+import zipkin.reporter.Encoder;
+import zipkin.reporter.internal.AwaitableCallback;
+
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KinesisSenderTest {
+  @Rule public MockWebServer server = new MockWebServer();
+
+  // Kinesis sends data in CBOR format
+  ObjectMapper mapper = new ObjectMapper(new CBORFactory());
```

Comment:
first I've ever seen this.. neat! https://github.com/FasterXML/jackson-dataformats-binary/tree/master/cbor